sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/JacksonVersion.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.jackson;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * Provides information about Jackson package versions used and detects potential issues.
+ */
+final class JacksonVersion {
+    private SemanticVersion annotationsVersion;
+    private SemanticVersion coreVersion;
+    private SemanticVersion databindVersion;
+    private SemanticVersion xmlVersion;
+    private SemanticVersion jsr310Version;
+
+    private static final String ANNOTATIONS_PACKAGE_NAME = "jackson-annotations";
+    private static final String CORE_PACKAGE_NAME = "jackson-core";
+    private static final String DATABIND_PACKAGE_NAME = "jackson-databind";
+    private static final String XML_PACKAGE_NAME = "jackson-dataformat-xml";
+    private static final String JSR310_PACKAGE_NAME = "jackson-datatype-jsr310";
+
+    private static final SemanticVersion MIN_SUPPORTED_VERSION = SemanticVersion.parse("2.10.0");
+    private static final SemanticVersion MAX_SUPPORTED_VERSION = SemanticVersion.parse("2.13.0-rc1");
+
+    private static final String AZURE_CORE_VERSION = CoreUtils
+        .getProperties("azure-core.properties")
+        .getOrDefault("version", SemanticVersion.UNKNOWN_VERSION);
```

Comment:
In time we should find a centralised location for this property so that it can be used in other places too.